### DISCUSSION TITLE
npm installing from repo with #main

### DISCUSSION THREAD

**Discussion by Dorian-Grim** [Orange-OpenSource/Orange-Boosted-Bootstrap#2470] [Q&A] [answered]
I am banging my head on the wall here, I can't figure this out. I am missing something but can't figure out what. So here's how it starts. 

The repo has been update with #2223. So I want to benefit asap from the new code and I go ahead and do an `npm install "https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap.git#main"`. Cli outputs it has updated my boosted package. When I npm start my app (react) and quickly change the `<html>` attribute `data-bs-theme` to `dark`, I see that it doesn't use the theme from [the preview page](https://deploy-preview-2223--boosted.netlify.app/)

On my dev env I start investigating:

1. inspect sources in node_modules
when I go here, I see the changes being pulled from the repo. More exactly, I quickly check https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/8ea14478475019858119c0679e71c17fb3bb4e7f/scss/_alert.scss#L81 and see if the line corresponds with the repo
3. inspect sources in browser
here however when I go to inspect, sources, open `.\node_modules\boosted\scss\_alert.scss` 
![image](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/24250692/7c78a50c-b307-45fd-898f-e4d95d4c2c5a)

I say, ok, it's a cache problem. I start and remove every cache I can think of possible. From the browser and from npm. Whatever I do, browser code (line 81) is still not updated to #2223 code

At this point I'm thinking, ok, my env is messed up. I go make a fresh, vm, install windows on it and then:

1. Install npm
2. I run `cd test-folder` then run `npx create-react-app my-app`
3. install boosted with `https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap.git#main`
4. I go into `my-app/node_modules/boosted/scss/_alert.scss` line `81` to check the #2223 mod. **It is correct, desired code**
5. I open `my-app/src/index.js`
6. I add `import "boosted/dist/css/boosted.min.css"`
7. I run `npm start`
app starts and opens
8. I inspect in browser and go to sources `.\node_modules\boosted\scss\_alert.scss` 
9. Line 81 is not using new code 🤦‍♂️ 

I have literally no idea why, how, where it's fetching the old code from.

Please help with anything 🙏

**Comment by julien-deramond** [comment, marked as answer]
Hey @Dorian-Grim 
When you're installing https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap.git#main, you gather the files from our current _main_ branch, so `my-app/node_modules/boosted/scss/_alert.scss` contains the new stuff.

**However**, we don't regenerate what's inside `dist` until there's a new release. So you have all the new Scss files, but when you `import "boosted/dist/css/boosted.min.css"` you still get the build from v5.3.2 (without the new modifications).
As a proof, you won't find any `data-bs-theme=dark` in `my-app/node_modules/boosted/dist/css/boosted.css` (or `.min.css`).

I think that if you want to achieve that, you have to recompile the CSS file from the Scss files contained in `my-app/node_modules/boosted/scss/` with Sass (more info here https://boosted.orange.com/docs/5.3/customize/sass/).

**Reply by Dorian-Grim** [reply]
Thank you very much for this!

1. I did `npm install -g sass`
2. deleted `boosted.css` and `boosted.css.map` from `node_modules/boosted/dist/css/`
3. from `my-app` folder issued a `sass --watch node_modules/boosted/scss/boosted.scss node_modules/boosted/dist/css/boosted.css`
4. ran `npm start`

I can now benefit from the `data-bs-theme="dark"`
